utilix/rundb.py

```diff
@@ -469,6 +388,28 @@ def delete_mc_document(self, document):
         url = '/mc/documents/'
         return self._delete(url, data=doc)
 
+    # TODO need to be careful about versioning!!!
+    def download_file(self, filename, save_dir='./', force=False):
```

Comment:
Does this actually sort of overwrite https://github.com/XENONnT/straxen/blob/master/straxen/mongo_storage.py#L278? After some careful discussions with Chris I made it such to pull the right files and do the right things with simultaneous read/writes & versions. I'm not sure what the best approach is but would this lead to two peaces of code doing something similar but not quite the same? 

I'm also not entirely confident I understand how the API works so my question might be based on a misunderstanding from my side.

Reply:
Hi Joran, yeah it is basically an exact copy but going through the API instead. The version stuff should be the same as straxen (grabs latest version), but we could add here the tempfile step like what straxen is doing to prevent the simultaneous write. I think that's a good idea. 

Then there's the question about having two pieces of code doing the same thing. We have many such examples running around that do not use the API but doing things that the API can do. I don't think it's a big problem, though maybe this one is unique in that it is being used to download data and not just do database interaction. If we don't want this then I think we should move the gridfs interface from straxen into utilix. It really is a database interaction so I feel it should be here anyway, to be honest.

Reply:
okay, clear, than I understand the rationale. 

> If we don't want this then I think we should move the gridfs interface from straxen into utilix. It really is a database interaction so I feel it should be here anyway, to be honest.

It's definitely in the gray area. I'm probably biased and do think it belongs to strax (perhaps more than straxen), the same way CMT is part of strax(en). Nevertheless, let's see, it is not a billion dollar question that is super important.

Reply:
Totally agree this is not that important, but let me get on my soapbox still for one minute 😉 

CMT is used to make corrections to the data so I definitely agree it belongs in strax(en) like it is now. This gridfs interface, though originally proposed specifically to use in conjunction with CMT, is different because it's much more general in scope. We are storing e.g. MC configuration there too, and could imagine other files being stored there as well that aren't necessarily part of loading/analyzing the data (maybe configuration or inputs for data management or our stats packages?). 

The whole idea of utilix is to be the package that handles common operations like this so that we don't get convoluted dependency graphs. If everything goes through strax or straxen we are following the example of how we used pax in XENON1T which was a total mess by the end (IMHO). Though of course gridfs is already a huge improvement over keeping all those files in a directory of pax. 

In my view, if an analyst is not working directly with strax-processed data, then they shouldn't have to import strax(en) just to get access to a database function. We should have a package that focuses on e.g. database operations (utilix) that then strax(en) can import from in order to do its goal of processing/analyzing data.